src/apps/api/tests/test_competitions.py

```diff
@@ -0,0 +1,52 @@
+import json
+
+from django.urls import reverse
+from rest_framework.test import APITestCase
+
+from competitions.models import CompetitionParticipant
+from factories import UserFactory, CompetitionFactory, CompetitionParticipantFactory, PhaseFactory, LeaderboardFactory, \
+    ColumnFactory
+
+
+class CompetitionParticipantTests(APITestCase):
+    def setUp(self):
+        self.creator = UserFactory(username='creator', password='creator')
+        self.other_user = UserFactory(username='other_user', password='other')
+        self.comp = CompetitionFactory(created_by=self.creator)
+        PhaseFactory(competition=self.comp)
+        self.leaderboard = LeaderboardFactory(competition=self.comp)
+        ColumnFactory(leaderboard=self.leaderboard)
+
+    def _prepare_competition_data(self, url):
+        # data = CompetitionSerializer(comp).data
+        # data.pop('id')
+        # data.pop('logo')
+        # return data
+        resp = self.client.get(url)
+        data = resp.data
+        data.pop('id')
+        data.pop('logo')
+
+        # Just get the key from the task and pass that instead of the object
+        data["phases"][0]["tasks"] = [data["phases"][0]["tasks"][0]["key"]]
+        return data
+
+    # TODO: Do we have competition permissions tests?
+    # def test_cant_edit_someone_elses_competition?
```

Comment:
to my knowledge, we don't have any tests that explicitly verify this code is working
https://github.com/codalab/competitions-v2/blob/935a4df38a94bee22b65214bf3a935eae05229c1/src/apps/api/views/competitions.py#L80

Reply:
✔️ made #299 for this